### PR TITLE
Add IsManagerAsync method

### PR DESCRIPTION
**Pull request by VLiamin**

### CODE REVIEW

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs [hunk 1]
```diff
@@ -65,5 +66,13 @@ public interface IAccessValidator
     /// <param name="rightIds">Ids of the rights.</param>
     /// <returns>True, if there's a UserId-RightId pair for at least one of requested rights in the database. False otherwise.</returns>
     Task<bool> HasAnyRightAsync(Guid? userId, bool includeIsAdminCheck, params int[] rightIds);
+
+    /// <summary>
+    /// Checks whether the user is mamager or not.
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs]: manager))

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs [hunk 2]
```diff
@@ -148,6 +157,38 @@ public async Task<bool> IsAdminAsync(Guid? userId = null)
...
+            ICheckDepartmentManagerRequest.CreateObj(userId, entityId),
+            timeout: 5000);
+          break;
+        default: 
+          return false;
+      }
+
+      if (!result.IsSuccess())
+      {
+        _logger.LogWarning("Failed to send request to CheckEntityManager via the broker.");
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs]: May be add to message source (Project/Department)?

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs (hunk 1 above)
Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/Interfaces/IAccessValidator.cs]: done

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs (hunk 2 above)
Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs]: done

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckDepartmentManagerRequest.cs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Requests
+{
+  public interface ICheckDepartmentManagerRequest
+  {
+    Guid UserId { get; }
+
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckDepartmentManagerRequest.cs:8]: remove empty line

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckProjectManagerRequest.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Requests
+{
+  public interface ICheckProjectManagerRequest
+  {
+    Guid UserId { get; }
+
```
Comment (vorobevamv) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckProjectManagerRequest.cs:8]: remove empty line

**Review by vorobevamv — APPROVED** [review]

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs [hunk 5]
```diff
@@ -44,11 +49,15 @@ private async Task<bool> IsUserAdminAsync(Guid userId)
...
-      IRequestClient<ICheckUserAnyRightRequest> requestClientAR)
+      IRequestClient<ICheckUserAnyRightRequest> requestClientAR,
+      IRequestClient<ICheckProjectManagerRequest> requestClientPM,
+      IRequestClient<ICheckDepartmentManagerRequest> requestClientDM)
     {
       _requestClientCR = requestClientCR;
       _requestClientUS = requestClientUS;
       _requestClientAR = requestClientAR;
+      _requestClientPM = requestClientPM;
+      _requestClientDM = requestClientDM;
```
Comment (ndoriomedov) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs]: rename to _rcCheckRights, etc

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs [hunk 6]
```diff
@@ -148,6 +157,38 @@ public async Task<bool> IsAdminAsync(Guid? userId = null)
...
+      Guid userId = _httpContext.GetUserId();
+
+      Response<IOperationResult<bool>> result = null;
+
+      switch (managerSource)
+      {
+        case ManagerSource.Project:
+          result = await _requestClientPM.GetResponse<IOperationResult<bool>>(
+            ICheckProjectManagerRequest.CreateObj(userId, entityId),
+            timeout: 5000);
```
Comment (ndoriomedov) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs]: maybe use requestHandler?

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs [hunk 7]
```diff
@@ -148,6 +157,38 @@ public async Task<bool> IsAdminAsync(Guid? userId = null)
       }
 
       return await IsUserAdminAsync(userId.Value);
+    }
+
+    public async Task<bool> IsManagerAsync(ManagerSource managerSource, Guid entityId)
```
Comment (ndoriomedov) [inline src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs:136]: dont u need /// <inheritdoc/> on 161?

Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs:136]: no

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckDepartmentManagerRequest.cs (hunk 3 above)
Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckDepartmentManagerRequest.cs:8]: This is similar to the existing request.

File: src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckProjectManagerRequest.cs (hunk 4 above)
Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/Requests/ICheckProjectManagerRequest.cs:8]: This is similar to the existing request.

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs (hunk 6 above)
Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs]: done

File: src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs (hunk 5 above)
Comment (VLiamin) [reply src/Kernel.BrokerSupport/AccessValidatorEngine/AccessValidator.cs]: done

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**